Add overloads to InvokeAsync with a return value

Jcparkyn opened the pull request:
Closes #1296

## Pull request description
Didn't include tests since there's essentially no logic changes here and the second overload already doesn't have any, but I can add them if wanted.

### PR meta checklist
- [x] Pull request is targeted at `main` branch for code   
  or targeted at `stable` branch for documentation that is live on bunit.dev.
- [x] Pull request is linked to all related issues, if any.
- [x] I have read the _CONTRIBUTING.md_ document.

### Code PR specific checklist
- [x] My code follows the code style of this project and AspNetCore coding guidelines.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [x] I have updated the appropriate sub section in the _CHANGELOG.md_.
- [ ] I have added, updated or removed tests to according to my changes.
  - [x] All tests passed.

Jcparkyn commented:
@dotnet-policy-service agree

linkdotnet commented:
Thanks for filing the PR @Jcparkyn !
It might be worth extending the documentation, shedding some light on the new use cases you enabled.

linkdotnet commented:
Hey @Jcparkyn,

if you have any documentation locally - just push it. The build server can take care of it.
Otherwise, we can also flatten out some rough edges.

linkdotnet commented:
@Jcparkyn  the docs should be totally fixed.

Jcparkyn commented:
Thanks @linkdotnet, just hadn't got around to writing it yet 😀.

Do you think it's worth adding a full example, or just a sentence or two? I'd think that most people who need this feature would discover it from the existing docs - maybe not for the `Task` use case though, since that's a bit more subtle.

Jcparkyn commented:
Added examples for both use cases, feel free to shorten it if you think they're not necessary. Also fixed a couple of incorrect line numbers in the docs.

By the way, the docs build and run properly now, but I think the readme instructions need updating again to use the `docfx` CLI.

linkdotnet commented on docs/site/docs/interaction/trigger-renders.md:
```diff
@@ -19,7 +19,7 @@ Let's look at how to use each of these methods to cause a re-render.
 The [`Render()`](xref:Bunit.RenderedComponentRenderExtensions.Render``1(Bunit.IRenderedComponentBase{``0})) method tells the renderer to re-render the component, i.e. go through its life-cycle methods (except for `OnInitialized()` and `OnInitializedAsync()` methods). To use it, do the following:
 
-[!code-csharp[](../../../samples/tests/xunit/ReRenderTest.cs?start=17&end=23&highlight=5)]
+[!code-csharp[](../../../samples/tests/xunit/ReRenderTest.cs?start=16&end=22&highlight=5)]
```
Nice - thanks!

linkdotnet commented on docs/site/docs/interaction/trigger-renders.md:
```diff
@@ -51,9 +51,25 @@ Let’s look at an example of this, using the `<Calc>` component listed below:
+
+### Advanced use cases
+
+In certain cases, the method you want to call might also have a return value, like the example below.
```
The passage is good. 

The whole documentation is in a neutral tone (not we, or you form). To align this, this part should also be in a more neutral form.

For example:

> In certain cases, the method you want to call might also have a return value, like the example below.

to:

> In some scenarios, the method being invoked may also yield a return value, as demonstrated in the following example.

Or this 

> You can test this exactly the same as before, but use the return value from `InvokeAsync()`:

to:

> Testing this scenario follows the same procedure as previously outlined, with the addition of utilizing the return value from InvokeAsync():

linkdotnet commented:
> By the way, the docs build and run properly now, but I think the readme instructions need updating again to use the docfx CLI.

Yes, you are right, and we missed updating this again :D 
I will do that later today.!

linkdotnet commented:
LGTM! Thanks for your efforts and time @Jcparkyn
